Sources/WalletConnectSign/Engine/Common/ApproveEngine.swift

```diff
@@ -82,7 +81,15 @@ final class ApproveEngine {
         let proposeResponse = SessionType.ProposeResponse(relay: relay, responderPublicKey: selfPublicKey.hexRepresentation)
         let response = JSONRPCResponse<AnyCodable>(id: payload.wcRequest.id, result: AnyCodable(proposeResponse))
 
+        guard var pairing = pairingStore.getPairing(forTopic: payload.topic) else {
+            throw Errors.pairingNotFound
+        }
+
         try await networkingInteractor.respond(topic: payload.topic, response: .response(response), tag: payload.wcRequest.responseTag)
+
+        try pairing.updateExpiry()
```

Comment:
we also extending pairing in `handleProposeResponse`. So in approve method we extending for wallet, and in handleProposeResponse we extending for dapp, right?

Reply:
correct